pkg/cmd/cli/cmd/env.go

```diff
@@ -260,7 +265,19 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 				return nil
 			}
 			for _, c := range containers {
+				oldEnv := c.Env
+				for _, env := range remove {
+					_, ok := findEnv(c.Env, env)
+					if !ok {
+						infosMsgs[i] = append(infosMsgs[i], fmt.Sprintf("environment variable %q not found in %s container\n", env, c.Name))
```

Comment:
if user is updating the envars in a pod, by default it will update all the containers. For that reason user should specify `--containers` flag I guess.